### PR TITLE
Fix/network performance

### PR DESCRIPTION
**Pull request by aensidhe**
it seems that tcp_nodelay should be enabled.

### CODE REVIEW

File: src/progaudi.tarantool/LogicalConnection.cs [hunk 1]
```diff
@@ -157,30 +142,31 @@ public static byte[] ReadFully(Stream input)
...
+            //catch
+            //{
+            //    _logWriter?.WriteLine($"Request with requestId {requestId} failed, header:\n{ToReadableString(headerBuffer)} \n body: \n{ToReadableString(bodyBuffer)}");
+            //    Dispose();
+            //    throw;
+            //}
+            //finally
+            //{
+            //    _physicalConnectionLock.ExitWriteLock();
+            //}
```
Comment (roman-kozachenko) [inline src/progaudi.tarantool/LogicalConnection.cs]: Please delete commented code.

File: src/progaudi.tarantool/ResponseReader.cs [hunk 2]
```diff
@@ -35,13 +35,43 @@ internal class ResponseReader : IResponseReader
...
+
+            try
+            {
+                _pendingRequestsLock.EnterWriteLock();
+
+                _clientOptions.LogWriter?.WriteLine("Cancelling all pending requests and setting faulted state...");
+
+                foreach (var response in _pendingRequests.Values)
+                {
+                    response.SetException(new InvalidOperationException("Can't read from physical connection."));
```
Comment (roman-kozachenko) [inline src/progaudi.tarantool/ResponseReader.cs]: Maybe something about Disposing? "Connection closed"?

File: src/progaudi.tarantool/ResponseWriter.cs [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+            lock (_lock)
+            {
+                if (_buffer.Count == 0)
+                    _newRequestsAvailable.Reset();
+            }
+
+            _physicalConnection.Flush();
+        }
+
+        private Tuple<ArraySegment<byte>, ArraySegment<byte>> GetRequest()
```
Comment (roman-kozachenko) [inline src/progaudi.tarantool/ResponseWriter.cs]: I think we should use one placing convention for GetRequest() and WriteBuffer() methods.

File: docker-compose.tests.yml [hunk 4]
```diff
@@ -1,10 +1,13 @@
 version: '2'
 
 services:
+
   tarantool-client:
     build: .
     command: /app/scripts/test-netcore.sh
     environment:
         TARANTOOL_REPLICATION_SOURCE: "tarantool:3301"
+        TEAMCITY_PROJECT_NAME: 123
```
Comment (aensidhe) [inline docker-compose.tests.yml]: Removed.

File: docker-compose.tests.yml [hunk 5]
```diff
@@ -1,10 +1,13 @@
...
     links:
-        - tarantool
\ No newline at end of file
+        - tarantool
+        - redis
```
Comment (aensidhe) [inline docker-compose.tests.yml]: Yes, for comparison. When we'll create benchmarks, we'll remove these tests.

**Pull request merged**